apis/dns/v1alpha1/resource_record_set_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ResourceRecordSetParameters define the desired state of a ResourceRecordSet
+type ResourceRecordSetParameters struct {
+	// Managed zone name that this ResourceRecordSet will be created in.
+	ManagedZone string `json:"managedZone"`
+
+	// Identifies what kind of resource this is.
+	//
+	// +kubebuilder:validation:Enum=dns#resourceRecordSet
+	Kind string `json:"kind"`
+
+	// The identifier of a supported record type.
+	//
+	// +kubebuilder:validation:Enum=A;AAAA;CAA;CNAME;DNSKEY;DS;IPSECKEY;MX;NAPTR;NS;PTR;SPF;SRV;SSHFP;TLSA;TXT
+	Type string `json:"type"`
+
+	// Number of seconds that this ResourceRecordSet
+	// can be cached by resolvers.
+	TTL int64 `json:"ttl"`
+
+	// List of ResourceRecord datas as defined in
+	// RFC 1035 (section 5) and RFC 1034 (section 3.6.1)
+	RRDatas []string `json:"rrdatas"`
+
+	// List fo Signature ResourceRecord datas, as
+	// defined in RFC 4034 (section 3.2).
+	SignatureRRDatas *[]string `json:"signatureRrdatas,omitempty"`
```

Comment:
Given slices are reference types, do we really need to define it as a pointer to make it optional?

Reply:
Makes sense. So I should remove this pointer and add the `// +optional` (I forgot it), right?

Reply:
Yes